lib/Feature/FeatureModelParser.cpp

```diff
@@ -225,4 +252,361 @@ std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)> FeatureModelXmlParser::parseDoc() {
+      Name = readUntil(To, ' ', Pos + 1);
+
+      // Note that we ignore the ID and use the name of the feature
+      // as unique identifier.
+      if (Name.find_first_of('(') != std::string::npos) {
+        Name = readUntil(Name, '(', 0);
+      }
+
+      // Remove the cardinality
+      if (Name.find_first_of('[') != std::string::npos) {
+        Name = readUntil(Name, '[', 0);
+      }
+
+      // If there is no name, provide an artificial one
+      if (Name.empty()) {
+        OrGroupCounter++;
+        Name = "group_" + std::to_string(OrGroupCounter);
+      }
+
+      // Create the feature
+      FMB.makeFeature<BinaryFeature>(Name, Opt);
+      IndentationToParentMapping[CurrentIndentationLevel] = Name;
+
+      // Add parent from the upper indentation level if there is one
+      if (LastIndentationLevel != -1 && CurrentIndentationLevel == RootIndentation) {
+        std::cerr << "Only one feature can be root and have the same indentation as root." << std::endl;
+        return false;
+      }
+
+      if (LastIndentationLevel != -1) {
+        auto Parent = IndentationToParentMapping.find(CurrentIndentationLevel - 1);
+        assert(Parent != IndentationToParentMapping.end());
+        FMB.addParent(Name, Parent->second);
+      }
+
+      // Add the or group to the feature model if it is completely parsed
+      auto OrGroup = OrGroupMapping.find(CurrentIndentationLevel);
+      if (OrGroup != OrGroupMapping.end()) {
+        FMB.addRelationship(std::get<1>(OrGroup->second), std::get<2>(OrGroup->second), std::get<0>(OrGroup->second));
+        OrGroupMapping.erase(CurrentIndentationLevel);
+      }
+
+      // Remember the new or group parent if there is one
+      if (Cardinalities.has_value()) {
+        Relationship::RelationshipKind GroupKind = Relationship::RelationshipKind::RK_ALTERNATIVE;
+        if (std::get<1>(Cardinalities.value()) == UINT_MAX) {
+          GroupKind = Relationship::RelationshipKind::RK_OR;
+        }
+        OrGroupMapping[CurrentIndentationLevel] = std::tuple<string, Relationship::RelationshipKind, std::vector<string>>(Name, GroupKind, std::vector<string>());
+      }
+
+      // Add a child
+      OrGroup = OrGroupMapping.find(CurrentIndentationLevel - 1);
+      if (OrGroup != OrGroupMapping.end()) {
+        std::get<2>(OrGroup->second).push_back(Name);
+      }
+
+
+      LastIndentationLevel = CurrentIndentationLevel;
+    }
+
+    // Add the remaining or groups
+    for (auto & OrGroup : OrGroupMapping) {
+      FMB.addRelationship(std::get<1>(OrGroup.second), std::get<2>(OrGroup.second), std::get<0>(OrGroup.second));
+    }
+  }
+
+  return true;
+}
+
+/// This method extracts the cardinality from the given line.
+/// The cardinality is wrapped in square brackets (e.g., [1,1])
+/// \param StringToExtractFrom the string to extract the cardinality from
+/// \return
+std::optional<std::tuple<int, int>> FeatureModelSxfmParser::extractCardinality(const string& StringToExtractFrom) {
```

Comment:
```suggestion
std::optional<std::tuple<int, int>> FeatureModelSxfmParser::extractCardinality(const string& StringToExtractFrom) {
```
Can take a StringRef, see description bellow.